### DISCUSSION TITLE
Dynamic footer - Gotenberg v 7

### DISCUSSION THREAD

**Discussion by Malinovskypl** [gotenberg/gotenberg#1066] [Q&A]
I am using Gotenberg to generate PDFs from HTML content. My requirement is to have a footer that only appears at the bottom of the last page. However, I encountered the following issues:

CSS limitations:

@media print does not seem to be supported in the Gotenberg rendering engine.
Advanced CSS features such as @page:last, @bottom-center, position: running(), and element() are also not working as expected.
Dynamic content:

The number of rows in my table is dynamic, so I cannot manually divide content into pages or predict where the content ends.
Resulting issue:

The footer is either repeated on every page or does not appear at the desired position on the last page.
What I Tried
Below is an example of the HTML and CSS I am using. This setup attempts to position the footer dynamically on the last page.

HTML and CSS Example

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Footer Only on Last Page</title>
    <style>
        @page {
            size: A4;
            margin: 20mm;
        }

        body {
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 0;
        }

        .page {
            position: relative;
            width: 100%;
            height: 297mm; /* Full A4 height */
            box-sizing: border-box;
            padding: 20mm;
            page-break-after: always;
        }

        .content {
            height: calc(100% - 30mm); /* Reserve space for footer */
            box-sizing: border-box;
        }

        .footer {
            position: fixed;
            bottom: 0;
            left: 0;
            width: 100%;
            height: 30mm;
            background-color: #333;
            color: white;
            text-align: center;
            padding: 10px 0;
            box-sizing: border-box;
        }

        /* Hide footer on non-last pages */
        .page:not(:last-child) .footer {
            display: none;
        }

        /* Table styling */
        table {
            width: 100%;
            border-collapse: collapse;
        }

        table th, table td {
            border: 1px solid #ddd;
            padding: 8px;
        }
    </style>
</head>
<body>
    <div class="page">
        <div class="content">
            <h1>Dynamic Content</h1>
            <p>This is a dynamically generated table:</p>
            <table>
                <thead>
                    <tr>
                        <th>Column 1</th>
                        <th>Column 2</th>
                    </tr>
                </thead>
                <tbody>
                    <!-- Example: Dynamic rows -->
                    <?php for ($i = 1; $i <= 200; $i++) { ?>
                        <tr>
                            <td>Row <?php echo $i; ?> - Col 1</td>
                            <td>Row <?php echo $i; ?> - Col 2</td>
                        </tr>
                    <?php } ?>
                </tbody>
            </table>
        </div>
        <div class="footer">
            <p>This footer appears only on the last page.</p>
        </div>
    </div>
</body>
</html>
```

**Observed Behavior**
The footer appears on every page instead of just the last one.
Alternatively, the footer does not position itself correctly at the bottom of the last page.

**Expected Behavior**
The footer should appear only on the last page.
It should always be positioned at the bottom of the page, regardless of the content's height or dynamic size.

Does Gotenberg support @media print and advanced paged media features like @page:last, @bottom-center, or position: running()?
If not, what is the recommended way to dynamically add a footer only to the last page in Gotenberg?
Are there any workarounds or configurations specific to Gotenberg that can achieve this behavior?

**Comment by thomasender** [comment]
@Malinovskypl have you found a solution for rendering the footer only on the last page? I am trying to apply different styling on the last page, but no success yet.

**Reply by Malinovskypl** [reply]
Unfortunately I didn't find it, but maybe you'll have better luck
